app/hotels/src/allHotels/AllHotels.js

```diff
@@ -40,21 +44,39 @@ type Props = {|
   openSingleHotel: (searchParams: AvailableHotelSearchInput) => void,
   openLocationPicker: (location: string | null) => void,
   openGuestsModal: () => void,
+  onCityIdChange: (cityId: string | null) => void,
   search: SearchParams,
   coordinates: Coordinates | null,
   location: string,
   filter: FilterParams,
   currency: string,
   cityId: string | null,
+  checkin?: string,
+  checkout?: string,
 |};
 
 /**
  * We need to lookup city ID first and only after that we can search for all
  * hotels. This is why we use two nested query renderers.
  */
-class AllHotels extends React.Component<Props> {
+export class AllHotels extends React.Component<Props> {
   componentDidMount = () => {
-    this.validateCheckinDate();
+    const { checkin, checkout, search } = this.props;
+
+    if (checkin && checkout && !isDateBeforeToday(DateFormatter(checkin))) {
+      this.props.onSearchChange({
+        ...search,
+        checkin: DateFormatter(checkin).toDate(),
+        checkout: DateFormatter(checkout).toDate(),
+      });
+    } else {
+      this.validateCheckinDate();
+    }
+  };
+
+  componentWillUnmount = () => {
+    this.props.onLocationChange('');
+    this.props.onCityIdChange(null);
```

Comment:
I noticed that if you chose a location from the location picker, clicked back to homepage, then clicked to to hotels search, it still searched with the last picked location, which is not what you would expect. So therefore I added this calls on unmount